File: demo-convert-to/demo-merge-csv-use-python-csv/merge.py
#!/usr/bin/env python3


import csv

class FileCsv:

	_FilePath = None
	def getFilePath (self):
		return self._FilePath
	def setFilePath (self, val):
		self._FilePath = val
		return self

	_Data = []
	def getData (self):
		return self._Data
	def setData (self, val):
		self._Data = val
		return self

	def toArray (self):
		return self.getData()

class FileCsvReader(FileCsv):

	def load (self):
		with open(self.getFilePath()) as csv_file:
			cells = csv.reader(csv_file)

			self.setData([])

			for cell in cells:
				self.getData().append(cell)

		return self

class FileCsvWriter(FileCsv):

	def save (self):
		with open(self.getFilePath(), 'w') as csv_file:
			writer = csv.writer(csv_file)
			writer.writerows(self.getData())


class AppMerge:

	def run (self):

		source_csv = FileCsvReader().setFilePath('1.csv').load()
		target_csv = FileCsvReader().setFilePath('3.csv').load()

		source_data = source_csv.toArray()
		target_data = target_csv.toArray()


		for target_row, target_cols in enumerate(target_data): ## 從「列(row)」開始巡迴

			if target_row == 0: ## 不處理第一列
				continue


			for target_col, cell_target in enumerate(target_cols): ## 從「某列(row)」開始巡迴「欄(col)」

				if target_col >= 2: ## 不處理第三攔以後
					continue

				source_cell = source_data[target_row][target_col]
				target_data[target_row][target_col] = source_cell


		# print(target_data)

		final_csv = FileCsvWriter().setFilePath('final.csv').setData(target_data)

		final_csv.save() ## 把「target_data」寫到「final.csv」。

if __name__ == '__main__':
	AppMerge().run()

File: demo-convert-to/demo-merge-csv-use-php-fputcsv/merge.php
#!/usr/bin/env php
<?php

abstract class FileCsv {
	protected $_FilePath = NULL;
	public function getFilePath ()
	{
		return $this->_FilePath;
	}
	public function setFilePath ($val)
	{
		$this->_FilePath = $val;
		return $this;
	}

	protected $_Handle = NULL;
	public function getHandle ()
	{
		return $this->_Handle;
	}
	public function setHandle ($val)
	{
		$this->_Handle = $val;
		return $this;
	}

	protected $_Data = array();
	public function getData ()
	{
		return $this->_Data;
	}
	public function setData ($val)
	{
		$this->_Data = $val;
		return $this;
	}

	public function toArray ()
	{
		return $this->getData();
	}

	public function open ()
	{
		return $this;
	}

	public function close ()
	{
		$handle = $this->getHandle();

		if ($handle === NULL) {
			return;
		}
		fclose($handle);

		$this->setHandle(NULL);

		return $this;
	}



}

class FileCsvReader extends FileCsv {

	public function open ()
	{
		if (($this->_Handle = fopen($this->getFilePath(), 'r')) === FALSE) {
			return;
		}

		return $this;
	}

	public function load ($length=0, $delimiter=',', $enclosure='"', $escape="\\")
	{
		$this->open();

		while (($cols = $this->readRow()) !== FALSE) {
			array_push($this->_Data, $cols);
		}

		$this->close();

		return $this;
	}

	public function readRow ($length=0, $delimiter=',', $enclosure='"', $escape="\\")
	{
		// http://php.net/manual/en/function.fgetcsv.php
		return fgetcsv($this->getHandle(), $length, $delimiter, $enclosure, $escape);
	}


}

class FileCsvWriter extends FileCsv {

	public function open ()
	{
		if (($this->_Handle = fopen($this->getFilePath(), 'w')) === FALSE) {
			return;
		}

		return $this;
	}

	public function save ($delimiter=',', $enclosure='"', $escape="\\")
	{
		$this->open();

		foreach ($this->getData() as $cells) {
			$this->writeRow($cells, $delimiter, $enclosure, $escape);
		}

		$this->close();

		return $this;
	}

	public function writeRow ($cells=array(), $delimiter=',', $enclosure='"', $escape="\\")
	{
		// http://php.net/manual/en/function.fputcsv.php
		return fputcsv($this->getHandle(), $cells, $delimiter=',', $enclosure='"', $escape="\\");
	}

}


class AppMerge {
	protected $_Csv = array();

	public function run ()
	{
		$this->_Csv['source'] = (new FileCsvReader)
			->setFilePath(__DIR__ . '/1.csv')
			->load()
		;

		$source_data = $this->_Csv['source']->toArray();

		$this->_Csv['target'] = (new FileCsvReader)
			->setFilePath(__DIR__ . '/3.csv')
			->load()
		;

		$target_data = $this->_Csv['target']->toArray();


		//var_dump($source_data);
		//var_dump($target_data);

		foreach ($target_data as $target_row => $target_cols) {

			if ($target_row === 0) { // 不處理第一列
				continue;
			}

			foreach ($target_cols as $target_col => $target_cell) {

				if ($target_col >= 2) { // 不處理第三攔以後
					continue;
				}

				$source_cell = $source_data[$target_row][$target_col];

				$target_data[$target_row][$target_col] = $source_cell;
			}
		}

		$this->_Csv['final'] = (new FileCsvWriter)
			->setFilePath(__DIR__ . '/final.csv')
		;

		$this->_Csv['final']
			->setData($target_data)
			->save()
		;

		/*
		foreach ($this->_Csv as $csv) {
			$csv->close();
		}
		*/

	}

}


(new AppMerge)
	->run()
;
